wallet/src/de/schildbach/wallet/ui/dashpay/DashPayViewModel.kt

```diff
@@ -104,4 +113,43 @@ class DashPayViewModel(application: Application) : AndroidViewModel(application)
             platformRepo.doneAndDismiss()
         }
     }
+
+    //TODO: this can probably be simplified using coroutines
+    private fun deriveEncryptionKey(onSuccess: (KeyParameter) -> Unit, onError: (Exception) -> Unit) {
+        val walletApplication = WalletApplication.getInstance()
+        val backgroundThread = HandlerThread("background", Process.THREAD_PRIORITY_BACKGROUND)
+        backgroundThread.start()
+        val backgroundHandler = android.os.Handler(backgroundThread.looper)
+        val securityGuard = SecurityGuard()
+        val password = securityGuard.retrievePassword()
+        object : DeriveKeyTask(backgroundHandler, walletApplication.scryptIterationsTarget()) {
+            override fun onSuccess(encryptionKey: KeyParameter, wasChanged: Boolean) {
+                onSuccess(encryptionKey)
+            }
+
+            override fun onFailure(ex: KeyCrypterException) {
+                onError(ex)
+            }
+        }.deriveKey(walletApplication.wallet, password)
+    }
```

Comment:
https://github.com/dashevo/dash-wallet/blob/evonet-develop/wallet/src/de/schildbach/wallet/ui/dashpay/CreateIdentityService.kt#L400

We have a co-routine version here.

Reply:
I see that you point the Co-routine method in PlatformRepo.